Remove redundant tests

geoHeil opened the pull request:
resolves #139 


consider this a first version. There may be additional tests we want to remove.

Here the idea was to simplify this task by just reducing backend implementations - maybe this is not enough and we really do want to clean up the test cases.

danielgafni commented on tests/metadata_stores/test_resolve_update.py:
```diff
@@ -38,15 +42,339 @@
 
 
-# ============= FEATURE GRAPH CONFIGURATIONS =============
+def verify_lazy_sql_frames(store: MetadataStore, lazy_result: LazyIncrement):
```
I don't think you need to add this: we already make an assertion with `df.implementation` (either in this test or golden provenance tests); but I will take a closer look

geoHeil replied:
removed

danielgafni commented on tests/metadata_stores/conftest.py:
```diff
@@ -260,21 +260,41 @@ def case_clickhouse(
         return (ClickHouseMetadataStore, {"connection_string": clickhouse_db})
 
 
+class BasicStoreCases:
```
Yeah I like this change

geoHeil commented:
I removed the thing from your comment - (i.e. addressed all open comments) but you wrote: Looking good so I thought you are fine with merging it.

Sorry - if this was too fast. Should we revert it?

danielgafni commented:
Yeah I think so
